[BE] Make torch.nn.modules.* satisfy the docs coverage test

mikaylagawarecki opened the pull request:
Options to address the "undocumented python objects":

1. Reference the functions in the .rst via the torch.nn.modules namespace. Note that this changes the generated doc filenames / locations for most of these functions!
2. [Not an option] Monkeypatch `__module__` for these objects (broke several tests in CI due to `inspect.findsource` failing after this change)
3. Update the .rst files to also document the torch.nn.modules forms of these functions, duplicating docs.

#### [this is the docs page added](https://docs-preview.pytorch.org/pytorch/pytorch/158491/nn.aliases.html) 
This PR takes option 3 by adding an rst page nn.aliases that documents the aliases in nested namespaces, removing all the torch.nn.modules.* entries from the coverage skiplist except 
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?)
- NonDynamicallyQuantizableLinear

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #158666
* #158654
* __->__ #158491



cc @albanD @mruberry @jbschlosser @walterddr

mikaylagawarecki committed:
[BE] Make torch.nn.modules.* satisfy the coverage test

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the coverage test"




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the coverage test"

Removed everything under `torch.nn.modules` from `coverage_ignore_classes` except
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?? lol)

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the coverage test"

Removed everything under `torch.nn.modules` from `coverage_ignore_classes` except
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?? lol)

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the coverage test"

Removed everything under `torch.nn.modules` from `coverage_ignore_classes` except
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?? lol)

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the docs coverage test"

Removed everything under `torch.nn.modules` from `coverage_ignore_classes` except
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?)
- NonDynamicallyQuantizableLinear

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.




cc albanD mruberry jbschlosser walterddr

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the docs coverage test"

Removed everything under `torch.nn.modules` from `coverage_ignore_classes` by documenting them as `torch.nn.modules.<Foo>.<Bar>` instead of `torch.nn.<Bar>` except
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?)
- NonDynamicallyQuantizableLinear

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.




cc albanD mruberry jbschlosser walterddr

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the docs coverage test"

Removed everything under `torch.nn.modules` from `coverage_ignore_classes` by documenting them as `torch.nn.modules.<Foo>.<Bar>` instead of `torch.nn.<Bar>` except
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?)
- NonDynamicallyQuantizableLinear

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.




cc albanD mruberry jbschlosser walterddr

[ghstack-poisoned]

janeyx99 commented on docs/source/nn.rst:
```diff
@@ -1,10 +1,15 @@
+   :maxdepth: 2
+   :hidden:
+
+   nn.aliases.rst
```
Wow it doesn't!
https://docs-preview.pytorch.org/pytorch/pytorch/158491/nn.html#containers

janeyx99 replied:
and this shows up! https://docs-preview.pytorch.org/pytorch/pytorch/158491/generated/torch.nn.modules.pixelshuffle.PixelShuffle.html#torch.nn.modules.pixelshuffle.PixelShuffle

when the main version this is unfound.

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the docs coverage test"

Options to address the "undocumented python objects":

1. Reference the functions in the .rst via the torch.nn.modules namespace. Note that this changes the generated doc filenames / locations for most of these functions!
2. [Not an option] Monkeypatch `__module__` for these objects (broke several tests in CI due to `inspect.findsource` failing after this change)
3. Update the .rst files to also document the torch.nn.modules forms of these functions, duplicating docs.

#### [this is the docs page added](https://docs-preview.pytorch.org/pytorch/pytorch/158491/nn.aliases.html) 
This PR takes option 3 by adding an rst page nn.aliases that documents the aliases in nested namespaces, removing all the torch.nn.modules.* entries from the coverage skiplist except 
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?)
- NonDynamicallyQuantizableLinear

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.




cc albanD mruberry jbschlosser walterddr

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Make torch.nn.modules.* satisfy the docs coverage test"

Options to address the "undocumented python objects":

1. Reference the functions in the .rst via the torch.nn.modules namespace. Note that this changes the generated doc filenames / locations for most of these functions!
2. [Not an option] Monkeypatch `__module__` for these objects (broke several tests in CI due to `inspect.findsource` failing after this change)
3. Update the .rst files to also document the torch.nn.modules forms of these functions, duplicating docs.

#### [this is the docs page added](https://docs-preview.pytorch.org/pytorch/pytorch/158491/nn.aliases.html) 
This PR takes option 3 by adding an rst page nn.aliases that documents the aliases in nested namespaces, removing all the torch.nn.modules.* entries from the coverage skiplist except 
- NLLLoss2d (deprecated)
- Container (deprecated)
- CrossMapLRN2d (what is this?)
- NonDynamicallyQuantizableLinear

This mostly required adding docstrings to `forward`, `extra_repr` and `reset_parameters`. Since forward arguments are already part of the module docstrings I just added a very basic docstring.




cc albanD mruberry jbschlosser walterddr

[ghstack-poisoned]
